Library/Homebrew/rubocops/components_order.rb

```diff
@@ -106,19 +106,26 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
             if on_macos_blocks.length == 1
               on_macos_block = on_macos_blocks.first
               child_nodes = on_macos_block.body.child_nodes
```

Comment:
Note: `BlockNode#body` doesn't always return what we might expect—see the [docs for `DefNode#body`](https://rubydoc.info/github/rubocop-hq/rubocop-ast/RuboCop/AST/DefNode#body-instance_method) (which also applies to `BlockNode#body`):

> **Note:** this can be either a `begin` node, if the method body contains multiple expressions, or any other node, if it contains a single expression.

This is why the `check_on_os_block_content` method found a bit farther down has this line:

```rb
child_nodes = on_os_block.body.begin_type? ? on_os_block.body.child_nodes : [on_os_block.body]
```

Reply:
For example, here's a commit on top of this PR's current commit tree that allows `if-else` statements inside `on_os` blocks: https://github.com/SeekingMeaning/brew/commit/38828e4570cad20db3dc972c8071fb875887235a

Reply:
Makes sense, thanks. Pushing your commit.